.travis.yml

```diff
@@ -1,69 +1,121 @@
 language: python
-python:
-- '3.6'
-- '3.7-dev'
 
-git:
-  depth: 200
-sudo: required
-dist: trusty
+matrix:
+  include:
+  - name: "Bionic python 3.7"
+    os: linux
+    dist: bionic
+    python: 3.7
+    env:
+      - REPO=https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
+      - OS=linux-64
+  - name: "trusty python 3.6"
+    os: linux
+    dist: trusty
+    python: 3.6
+    env:
+      - REPO=https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
+      - OS=linux-64
+  - name: "xenial python 3.7"
+    os: linux
+    dist: xenial
+    python: 3.7
+    env:
+      - REPO=https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
+      - OS=linux-64
+  - name: "xenial python 3.6"
+    os: linux
+    dist: xenial
+    python: 3.6
+    env:
+      - REPO=https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
+      - OS=linux-64
+  - name: "osx python 3.7"
+    os: osx
+    language: generic
+    env:
+    - REPO=https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
+    - TRAVIS_PYTHON_VERSION=3.7
```

Comment:
Variable TRAVIS_PYTHON_VERSION is not defined in all cases, is it normal ?

Reply:
yes, it is defined when it's the default platform language. However on Mac it is not possible to define a job with default language "python" (thus the `language: generic`). 
See for example:

https://docs.travis-ci.com/user/multi-os/#python-example-unsupported-languages